tests/NLog.UnitTests/LogToDiagnosticsTests.cs

```diff
@@ -0,0 +1,260 @@
+
+            Assert.Equal(1, mockTraceListener.Messages.Count);
+            Assert.Equal("NLog: Debug Logger1 Hello" + Environment.NewLine, mockTraceListener.Messages.First());
+        }
+
+        [Theory]
+        [InlineData(true, null)]
+        [InlineData(null, true)]
+        [InlineData(true, true)]
+        public void ShouldLogToDiagnosticsWhenInternalLogToDiagnosticsIsOnAndLogLevelIsInfo(bool? internalLogToDiagnostics, bool? logToDiagnostics)
+        {
+            var mockTraceListener = SetupTestConfiguration(LogLevel.Info, CreateMockTraceListener, internalLogToDiagnostics, logToDiagnostics);
+
+            InternalLogger.Info("Logger1 Hello");
+
+            Assert.Equal(1, mockTraceListener.Messages.Count);
+            Assert.Equal("NLog: Info Logger1 Hello" + Environment.NewLine, mockTraceListener.Messages.First());
+        }
+
+        [Theory]
+        [InlineData(true, null)]
+        [InlineData(null, true)]
+        [InlineData(true, true)]
+        public void ShouldLogToDiagnosticsWhenInternalLogToDiagnosticsIsOnAndLogLevelIsWarn(bool? internalLogToDiagnostics, bool? logToDiagnostics)
+        {
+            var mockTraceListener = SetupTestConfiguration(LogLevel.Warn, CreateMockTraceListener, internalLogToDiagnostics, logToDiagnostics);
+
+            InternalLogger.Warn("Logger1 Hello");
+
+            Assert.Equal(1, mockTraceListener.Messages.Count);
+            Assert.Equal("NLog: Warn Logger1 Hello" + Environment.NewLine, mockTraceListener.Messages.First());
+        }
+
+        [Theory]
+        [InlineData(true, null)]
+        [InlineData(null, true)]
+        [InlineData(true, true)]
+        public void ShouldLogToDiagnosticsWhenInternalLogToDiagnosticsIsOnAndLogLevelIsError(bool? internalLogToDiagnostics, bool? logToDiagnostics)
+        {
+            var mockTraceListener = SetupTestConfiguration(LogLevel.Error, CreateMockTraceListener, internalLogToDiagnostics, logToDiagnostics);
+
+            InternalLogger.Error("Logger1 Hello");
+
+            Assert.Equal(1, mockTraceListener.Messages.Count);
+            Assert.Equal("NLog: Error Logger1 Hello" + Environment.NewLine, mockTraceListener.Messages.First());
+        }
+
+        [Theory]
+        [InlineData(true, null)]
+        [InlineData(null, true)]
+        [InlineData(true, true)]
+        public void ShouldLogToDiagnosticsWhenInternalLogToDiagnosticsIsOnAndLogLevelIsFatal(bool? internalLogToDiagnostics, bool? logToDiagnostics)
+        {
+            var mockTraceListener = SetupTestConfiguration(LogLevel.Fatal, CreateMockTraceListener, internalLogToDiagnostics, logToDiagnostics);
+
+            InternalLogger.Fatal("Logger1 Hello");
+
+            Assert.Equal(1, mockTraceListener.Messages.Count);
+            Assert.Equal("NLog: Fatal Logger1 Hello" + Environment.NewLine, mockTraceListener.Messages.First());
+        }
+
+        /*[Fact]
+        public void ShouldThrowStackOverFlowExceptionWhenUsingNLogTraceListener()
```

Comment:
Please add the failing unit test with `[Ignore]`. I guess we won't expect a `StackOverflowException`? (so we need to alter this unit test)